src/subcommands/diff.rs

```diff
@@ -76,17 +99,97 @@ pub fn diff(
         .unwrap_or_else(|| {
             eprintln!("'{diff_bin}' process terminated without exit status.");
             config.error_exit_code
-        })
+        });
+
+    if code >= 2 {
+        let via_process_substitution =
+            |f: &Path| f.starts_with("/proc/self/fd/") || f.starts_with("/dev/fd/");
+        let is_git_diff = matches!(differ, Differ::GitDiff);
+        if is_git_diff
```

Comment:
Why not `#[derive(PartialEq)]` (and `Debug` for good measure)? Then just `==` is possible.

Reply:
Done, thanks.